libflux/include/influxdata/flux.h

```diff
@@ -1,19 +1,48 @@
 #ifndef _INFLUXDATA_FLUX_H
 #define _INFLUXDATA_FLUX_H
 
+#include <stddef.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 // flux_ast_t is the AST representation of a flux query.
 struct flux_ast_t;
 
+// flux_buffer_t is a reference to a byte-slice.
+struct flux_buffer_t {
```

Comment:
As an FYI, it will be just a bit more complicated to make a flatbuffer available to Go.  Since the flatbuffer builder builds from the end of the buffer backwards, there will be some unused space at the beginning of the buffer, but Go will still need a pointer to the first byte to pass to `Free`.  So the analogous data structure for flatbuffers will need an `offset` field.

Reply:
That's good to know. We'll figure out if we need that. The pointer is not always supposed to be freed as sometimes it's a reference and there isn't really a way to deal with that. We can also always return the offset as part of a reference variable that gets set by the C function equivalent or we'll find some way to return the offset.